### ISSUE TITLE
Tensorflow - Optimizers 

### ISSUE THREAD

**Issue by junxnone** [junxnone/tio#434]
- Optimizer 
- GradientDescentOptimizer 
- AdagradOptimizer 
- AdagradDAOptimizer 
- MomentumOptimizer 
- AdamOptimizer 
- FtrlOptimizer 
- RMSPropOptimizer

# Reference
- [ ] [An overview of gradient descent optimization algorithms](http://ruder.io/optimizing-gradient-descent/) --- [paper](https://arxiv.org/pdf/1609.04747.pdf)
- [ ] [Tensorflow 优化器Optimizer](https://blog.csdn.net/xierhacker/article/details/53174558)
- [ ] [关于深度学习优化器 optimizer 的选择，你需要了解这些](https://www.leiphone.com/news/201706/e0PuNeEzaXWsMPZX.html)

**Comment by junxnone** [comment]
#22 #435 #171 #438
